src/sql/platform/connection/common/connectionStatusManager.ts

```diff
@@ -49,6 +49,19 @@ export class ConnectionStatusManager {
 		return !!this.findConnection(id);
 	}
 
+	public changeConnectionUri(newId: string, oldId: string) {
+		let info = this.findConnection(oldId);
+		if (!info) {
+			this._logService.error(`No connection found associated with old URI : '${oldId}'`);
+		}
+		info.ownerUri = newId;
+		if (this._connections[newId]) {
+			this._logService.error(`New URI : '${newId}' is already in the connections list.`);
```

Comment:
This check should come before updating the info - since we don't want to update the URI if either case is true (also same as above, this should probably throw)